extra_data/components.py

```diff
@@ -176,6 +179,13 @@ def _select_trains(cls, data, mod_data_counts, min_modules):
     def train_ids(self):
         return self.data.train_ids
 
+    @property
+    def frames_per_train(self):
+        counts = set(self.frame_counts.unique()) - {0}
+        if len(counts) > 1:
+            raise ValueError(f"Varying number of frames per train: {counts}")
+        return counts.pop()
```

Comment:
It shouldn't be possible to create the object with no data in the first place, because of this check:

https://github.com/European-XFEL/EXtra-data/blob/ee3822c5b617a56495992666a986f0a521c23989/extra_data/components.py#L170-L172

But it might happen if you pass `min_modules=0`. I'll add an extra check in.